bigc/resources/products.py

```diff
@@ -8,7 +8,8 @@ class BigCommerceProductsAPI:
     def __init__(self, api_client: BigCommerceAPIClient):
         self._api = api_client
 
-    def all(self, *, include_variants: bool = False, include_custom_fields: bool = False) -> Iterator[dict]:
+    def all(self, *, include_variants: bool = False, include_custom_fields: bool = False,
+            extra_params: dict[str, str] = {}) -> Iterator[dict]:
```

Comment:
You should not pass a mutable as a default for a python argument, meaning that `{}` should never be a default for an argument. 

When you want an argument to default to an empty dictionary the pattern to use is 
```
def foo(bar=None)
    if bar is None
        bar = {}
    ...
```